### PR TITLE
[FIX] Round bhattacharayya

### PR DESCRIPTION
**Pull request by AndrejaKovacic**
##### Issue

Due to floating point precision,  bhattacharayya distance would sometimes return negative distances.

##### Description of changes
Rounding numbers to 13th digit. I think this is precise enough for our purposes, otherwise, we could use decimal class instead of floats.

##### Includes
- [X] Code changes

### CODE REVIEW

File: Orange/distance/distance.py [hunk 1]
```diff
@@ -750,8 +750,8 @@ def _bhattacharyya(a, b):
     a = _prob_dist(a)
     b = _prob_dist(b)
     if sp.issparse(a):
-        return -np.log(np.sum(np.sqrt(a.multiply(b))))
-    return -np.log(np.sum(np.sqrt(a * b)))
+        return np.clip(-np.log(np.sum(np.sqrt(a.multiply(b)))), 0, None)
+    return np.clip(-np.log(np.sum(np.sqrt(a * b))), 0, None)
```
Comment (janezd) [inline Orange/distance/distance.py]: I know I'm being an a..., I mean, annoying, but I would find it much nicer if you changed the code to something like

```
if sp.issparse(a):
    prod = a.multiply(b)
else:
    prod = a * b
return np.clip(-np.log(np.sum(np.sqrt(prod))
```

Or even `prod = a.multiply(a) if sp.issparse(a) else a * b`.

Comment (AndrejaKovacic) [reply Orange/distance/distance.py]: it does look nicer, i changed it.

Comment (markotoplak) [reply Orange/distance/distance.py]: Hmmm, is the condition even needed? Wouldn't `a.multiply(b)` work to with sparse and dense matrices?

Comment (janezd) [reply Orange/distance/distance.py]: You seem to think I haven't tried. :)

```
>>> import numpy as np
>>> a = np.arange(20).reshape(4, 5)
>>> a.multiply(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'numpy.ndarray' object has no attribute 'multiply'
```

There is `np.multiply`, but it doesn't work for sparse matrices.

Comment (AndrejaKovacic) [reply Orange/distance/distance.py]: No, numpy arrays don't have multiply function, only the numpy module has it.

**Pull request merged**